mxcube3/routes/SampleCentring.py

```diff
@@ -601,8 +601,29 @@ def accept_centring():
 
 @mxcube.route("/mxcube/api/v0.1/sampleview/centring/reject", methods=['PUT'])
 def reject_centring():
-    """
-    Reject the centring position.
-    """
+    """Reject the centring position."""
     mxcube.diffractometer.rejectCentring()
     return Response(status=200)
+
+
+@mxcube.route("/mxcube/api/v0.1/sampleview/moveto", methods=['PUT'])
+def pos_from_x_y():
```

Comment:
I think calculations in general should be delegated to underlying hardware objects, when a similar functionality exists, in order to keep a clean code in the Python server.

So I would do something like this (pseudo code, not tested):

    try:
        mxcube.diffractometer.moveToBeam()
    except AttributeError:
        # 2.2 and above
        mxcube.diffractometer.start_move_to_beam()